src/mixer/basetrackplayer.h

```diff
@@ -100,6 +107,10 @@ class BaseTrackPlayerImpl : public BaseTrackPlayer {
     ControlObjectSlave* m_pPreGain;
     ControlObjectSlave* m_pSpeed;
     ControlObjectSlave* m_pPitchAdjust;
+    QScopedPointer<ControlObjectSlave> m_pInputEnabled;
+    QScopedPointer<ControlObjectSlave> m_pPassthroughEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlStatus;
```

Comment:
Right, I'm saying this removeAt and event are a cost of using a smart
pointer which massively eclipses the cost of the hypothetical extra
instructions the smart pointer costs -- ie this discussion is silly.

On Wed, Jan 20, 2016, 11:43 PM Daniel Schürmann <notifications@github.com>
wrote:

> In src/mixer/basetrackplayer.h
> <https://github.com/mixxxdj/mixxx/pull/858#discussion_r50368688>:
>
> > @@ -100,6 +107,10 @@ class BaseTrackPlayerImpl : public BaseTrackPlayer {
> >      ControlObjectSlave* m_pPreGain;
> >      ControlObjectSlave* m_pSpeed;
> >      ControlObjectSlave* m_pPitchAdjust;
> > +    QScopedPointer<ControlObjectSlave> m_pInputEnabled;
> > +    QScopedPointer<ControlObjectSlave> m_pPassthroughEnabled;
> > +    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
> > +    QScopedPointer<ControlObjectSlave> m_pVinylControlStatus;
>
> Did you look at the source?
> This is the children delete Loop, once the widget has had children, the
> loop will run. The currentChildBeingDeleted guard, protects us from
> sending a QEvent::ChildRemoved in the childs destructor and from a
> QList::removeAt(), which
> are called in the smart pointer case.
>
> void QObjectPrivate::deleteChildren()
> {
>     Q_ASSERT_X(!isDeletingChildren, "QObjectPrivate::deleteChildren()", "isDeletingChildren already set, did this function recurse?");
>     isDeletingChildren = true;
>     // delete children objects
>     // don't use qDeleteAll as the destructor of the child might
>     // delete siblings
>     for (int i = 0; i < children.count(); ++i) {
>         currentChildBeingDeleted = children.at(i);
>         children[i] = 0;
>         delete currentChildBeingDeleted;
>     }
>     children.clear();
>     currentChildBeingDeleted = 0;
>     isDeletingChildren = false;
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/858/files#r50368688>.
>